libcxx/test/libcxx/utilities/meta/is_within_lifetime.verify.cpp

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03
+
+// <type_traits>
+
+// LWG4138 <https://cplusplus.github.io/LWG/issue4138>
+// std::is_within_lifetime shouldn't work when a function type is
+// explicitly specified, even if it isn't evaluated
+
+#include <type_traits>
+#include <cassert>
+
+#include "test_macros.h"
+
+void fn();
+
+int main(int, char**) {
+#ifdef __cpp_lib_is_within_lifetime
```

Comment:
Instead, here I would do the test unconditionally but I would mark it as

```
// UNSUPPORTED: clang-18, clang-19, gcc-14, apple-clang-16
```

We are used to removing these `UNSUPPORTED` annotations as we bump compiler requirements, so this "tech debt" will eventually go away naturally. Otherwise, this `#ifdef __cpp_lib_is_within_lifetime` will stay here forever (unless someone stumbles upon it and realizes that it can be removed, which is unlikely).